clients/devtools-entry.js

```diff
@@ -67,6 +68,70 @@ function listenForStatus(listenCallback) {
   log.events.addListener('status', listenCallback);
 }
 
+/**
+ * With just a trace, provide Lighthouse performance report
+ * From DevTools it'll look something like this
+ *
+    self.analyzeTrace({traceEvents: tEvents}, {url: 'http://page', device: 'desktop'})
+      .then(console.log)
+      .catch(console.warn);
+ *
+ * @param {LH.Trace} trace
+ * @param {{device: "mobile" | "desktop", url: string}} opts
+ * @return {Promise<LH.RunnerResult|undefined>}
+ */
+function analyzeTrace(trace, opts) {
+  const url = opts.url;
+
+  /** @type {LH.Config.Json} */
+  const configJSON = {
+    extends: 'lighthouse:default',
+    settings: {
+      onlyCategories: ['performance'],
+      output: ['json'],
+      formFactor: opts.device,
+      throttlingMethod: 'devtools', // can't do lantern right now, so need real throttling applied.
+      // In DevTools, emulation is applied _before_ Lighthouse starts (to deal with viewport emulation bugs). go/xcnjf
+      // As a result, we don't double-apply viewport emulation.
+      screenEmulation: {disabled: true},
+      traceBasedNetworkRecords: true,
+    },
+  };
+
+  // TODO: use FR's initializeConfig. it'll filter for navigation-y things.
+  const config = lighthouse.generateConfig(configJSON, {});
+  const runOpts = {url, config, computedCache: new Map()};
+
+  const gatherFn = () => {
+    /** @type {LH.DevtoolsLog} */
+    const fakeDtLogs = [];
+    fakeDtLogs.smuggledTrace = trace;
+
+    // /** @type {Partial<LH.Artifacts>} */
+    const artifacts = {
+      traces: {defaultPass: trace},
+      devtoolsLogs: {defaultPass: fakeDtLogs},
+
+      GatherContext: {gatherMode: 'navigation'}, // todo: timespan???
```

Comment:
yea whats up with this, perf panel can trigger a trace 1) with a page reload+navigation 2) on/off at user request. At least, old perf panel can, I assume new one too. So do we need to be told what kind of trace we are getting? Part of the api for this function?

Reply:
That's a good point. Tricky.. I'll get back to you on this.

Reply:
devtools ends up doing some work to only feed us data if there's a navigation in the main frame, so that ends up meaning we only handle the navigate case. go/glsmn LH literally wont get a chance to even parse it if there's not a trace event saying a new document is loading in the main frame.  So... that makes things fairly straightforward for us.

In the future, we can definitely allow a FR timespan case.